src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -58,14 +53,28 @@ public void storeMessage(Message message) {
 
     /** Stores the Location in Datastore. */
     public void storeLocation(Location location) {
+
         Entity locationEntity = new Entity("Location", location.getID().toString());
         locationEntity.setProperty("name", location.getName());
         locationEntity.setProperty("description", location.getDescription());
+        locationEntity.setProperty("imageUrl", location.getImageURL());
+        locationEntity.setProperty("imageLabels", location.getImageLabels());
+        locationEntity.setProperty("reviews", location.getReviews());
+        locationEntity.setProperty("idString", location.getID());
 
         datastore.put(locationEntity);
     }
 
-  /**
+    /** Retrieves Location Entity with specific id in Datastore. */
+    public Entity getLocation(String id) {
+
+        Query.Filter idFilter = new Query.FilterPredicate("idString", FilterOperator.EQUAL, id);
+        Query query = new Query("location").setFilter(idFilter);
+        PreparedQuery results = datastore.prepare(query);
+        return results.asSingleEntity();
+    }
+
```

Comment:
Is this extra line fine? @jonathantanmy

Reply:
Yes, this extra line is fine.

It's fine if you added or edited at least one of the adjacent lines, and want to add some space (which is true in this case). The time when it is bad is if you added or removed a blank line in some unrelated part of the file.